internal/cache/file/downloader/job_test.go

```diff
@@ -209,3 +208,73 @@ func (jt *jobTest) Test_failWhileDownloading() {
 	ExpectTrue(reflect.DeepEqual(jobStatus, notification2))
 	ExpectEq(true, ok2)
 }
+
+func (jt *jobTest) Test_updateFileInfoCache_UpdateEntry() {
+	// Add an entry into
+	fileInfoKey := data.FileInfoKey{
+		BucketName: storage.TestBucketName,
+		ObjectName: DefaultObjectName,
+	}
+	fileInfo := data.FileInfo{
+		Key:              fileInfoKey,
+		ObjectGeneration: jt.job.object.Generation,
+		FileSize:         jt.job.object.Size,
+		Offset:           0,
+	}
+	fileInfoKeyName, err := fileInfoKey.Key()
+	ExpectEq(nil, err)
+	_, err = jt.cache.Insert(fileInfoKeyName, fileInfo)
+	ExpectEq(nil, err)
+	jt.job.status.Offset = 1
+
+	err = jt.job.updateFileInfoCache()
+
+	ExpectEq(nil, err)
+	// confirm fileInfoCache is updated with new offset.
+	lookupResult := jt.cache.LookUp(fileInfoKeyName)
+	ExpectFalse(lookupResult == nil)
+	fileInfo = lookupResult.(data.FileInfo)
+	ExpectEq(1, fileInfo.Offset)
+	ExpectEq(jt.job.object.Generation, fileInfo.ObjectGeneration)
+	ExpectEq(jt.job.object.Size, fileInfo.FileSize)
+}
+
+func (jt *jobTest) Test_updateFileInfoCache_InsertNew() {
```

Comment:
Thinking loud - later when we start using overrideKey of lru, inserting a new entry will return an error.

Reply:
Added a comment to remove this test when we support updateValue in lru.